tests/test_data_tokenizers.py

```diff
@@ -0,0 +1,620 @@
+import pytest
+import random
+import collections
+import pickle
+from uuid import uuid4
+import os
+import unicodedata
+import tempfile
+from gluonnlp.data.tokenizers import WhitespaceTokenizer, MosesTokenizer, JiebaTokenizer,\
+    SpacyTokenizer, SubwordNMTTokenizer, YTTMTokenizer, SentencepieceTokenizer, \
+    HuggingFaceBPETokenizer, HuggingFaceByteBPETokenizer, HuggingFaceWordPieceTokenizer
+from gluonnlp.base import get_repo_url
+from gluonnlp.data import Vocab
+from gluonnlp.utils.misc import download
+
+
+EN_SAMPLES = ['Four score and seven years ago our fathers brought forth on this continent, '
+              'a new nation, conceived in Liberty, and dedicated to the proposition '
+              'that all men are created equal.',
+              'In spite of the debate going on for months about the photos of Özil with the '
+              'Turkish President Recep Tayyip Erdogan, he regrets the return of '
+              'the 92-match national player Özil.']
+DE_SAMPLES = ['Goethe stammte aus einer angesehenen bürgerlichen Familie; sein Großvater'
+              ' mütterlicherseits war als Stadtschultheiß höchster Justizbeamter der'
+              ' Stadt Frankfurt, sein Vater Doktor der Rechte und kaiserlicher Rat.',
+              '"Das ist eine Frage, die natürlich davon abhängt, dass man einmal ins '
+              'Gespräch kommt, dass man mit ihm auch darüber spricht, warum er das eine '
+              'oder andere offenbar so empfunden hat, wie das in seinem Statement niedergelegt'
+              ' ist", sagte Grindel im Fußball-Podcast "Phrasenmäher" der "Bild-Zeitung.']
+ZH_SAMPLES = ['苟活者在淡红的血色中，会依稀看见微茫的希望；真的猛士，将更奋然而前行。',
+              '参加工作，哈尔滨工业大学无线电工程系电子仪器及测量技术专业毕业。']
+
+SUBWORD_TEST_SAMPLES = ["Hello, y'all! How are you Ⅷ 😁 😁 😁 ?",
+                        'GluonNLP is great！！！!!!',
+                        "GluonNLP-Amazon-Haibin-Leonard-Sheng-Shuai-Xingjian...../:!@# 'abc'"]
```

Comment:
@leezu @eric-haibin-lin These are the test cases for tokenization. We test for 
- encode
- decode
- encode_with_offsets
- Whether the tokenizer is pickleble